chain/chain/src/flat_storage_resharder.rs

```diff
@@ -136,89 +136,56 @@ impl FlatStorageResharder {
-                // Send a request to schedule the execution of `shard_catchup_task` for this shard.
-                self.sender
-                    .flat_storage_shard_catchup_sender
-                    .send(FlatStorageShardCatchupRequest { resharder: self.clone(), shard_uid });
+                info!(target: "resharding", ?shard_uid, ?resharding_status, "resuming flat storage shard catchup");
+                match self.shard_catchup_task(shard_uid) {
+                    // All good.
+                    FlatStorageReshardingTaskResult::Successful { .. } => {}
+                    FlatStorageReshardingTaskResult::Failed => {
+                        panic!("impossible to recover from a flat storage shard catchup failure!")
+                    }
+                    // The task has been cancelled. Nothing else to do.
+                    FlatStorageReshardingTaskResult::Cancelled => {}
+                }
             }
         }
         Ok(())
     }
 
-    /// Starts the event of splitting a parent shard flat storage into two children.
-    ///
-    /// This method is resilient to chain forks and concurrent resharding requests. First of all, if
-    /// a resharding event has started, no other split shard events will be accepted. Second, if a
-    /// resharding event is already scheduled, the new event will extend the old one (if they are
-    /// the same type) or replace it.
-    fn split_shard(
+    /// The preprocessing step sets the appropriate flat storage status for the parent and children shards.
+    fn split_shard_task_preprocessing(
         &self,
-        split_params: ReshardingSplitShardParams,
-        shard_layout: &ShardLayout,
-    ) -> Result<(), Error> {
-        let ReshardingSplitShardParams {
-            parent_shard,
-            left_child_shard,
-            right_child_shard,
-            resharding_block,
-            ..
-        } = split_params;
-        info!(target: "resharding", ?split_params, "initiating flat storage shard split");
-
-        self.check_new_event_is_allowed()?;
-
-        // Get all resharding blocks, if any, from the existing scheduled split shard,
-        // and merge them with the new resharding block.
-        let mut resharding_blocks = vec![resharding_block];
-        if let Some(FlatStorageReshardingEventStatus::SplitShard(parent, params, _)) =
-            self.resharding_event()
-        {
-            if parent_shard == parent
-                && params.left_child_shard == left_child_shard
-                && params.right_child_shard == right_child_shard
-                && params.shard_layout == *shard_layout
-            {
-                resharding_blocks.extend(params.resharding_blocks);
-            }
-        };
-
-        // Cancel any scheduled, not yet started event.
-        self.cancel_scheduled_event();
-
+        parent_shard: ShardUId,
+        split_params: &ParentSplitParameters,
```

Comment:
It always baffles me that parent shard isn't a part of the parameters

Reply:
I agree.. but I didn't want to make any changes to the `ParentSplitParameters` as this was already a very large PR.